packages/salesforcedx-vscode-core/src/commands/isvdebugging/bootstrapCmd.ts

```diff
@@ -64,52 +78,278 @@ export class IsvDebugBootstrapExecutor extends SfdxCommandletExecutor<{}> {
+      )
+      .withArg('force:mdapi:retrieve')
+      .withFlag('-r', this.relativeMetdataTempPath)
+      .withFlag('-k', this.relativeApexPackageXmlPath)
+      .withFlag('-u', data.sessionId)
+      .build();
+  }
+
+  public buildMetadataApiConvertOrgSourceCommand(
+    data: IsvDebugBootstrapConfig
+  ): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(
+        nls.localize('isv_debug_bootstrap_step4_convert_org_source')
+      )
+      .withArg('force:mdapi:convert')
+      .withFlag('-r', path.join(this.relativeMetdataTempPath, 'unpackaged'))
+      .withFlag('-d', 'force-app')
+      .build();
+  }
+
+  public buildPackageInstalledListAsJsonCommand(
+    data: IsvDebugBootstrapConfig
+  ): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(
+        nls.localize('isv_debug_bootstrap_step5_list_installed_packages')
+      )
+      .withArg('force:package:installed:list')
+      .withFlag('-u', data.sessionId)
+      .withArg('--json')
+      .build();
+  }
+
+  public buildRetrievePackagesSourceCommand(
+    data: IsvDebugBootstrapConfig,
+    packageNames: string[]
+  ): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(
+        nls.localize('isv_debug_bootstrap_step6_retrieve_packages_source')
+      )
+      .withArg('force:mdapi:retrieve')
+      .withFlag('-r', this.relativeMetdataTempPath)
+      .withFlag('-p', packageNames.join(','))
+      .withFlag('-u', data.sessionId)
+      .build();
+  }
+
+  public buildMetadataApiConvertPackageSourceCommand(
+    packageName: string
+  ): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(
+        nls.localize(
+          'isv_debug_bootstrap_step7_convert_package_source',
+          packageName
+        )
+      )
+      .withArg('force:mdapi:convert')
+      .withFlag(
+        '-r',
+        path.join(this.relativeMetdataTempPath, 'packages', packageName)
+      )
+      .withFlag('-d', path.join('packages', packageName))
+      .build();
+  }
+
+  public parsePackageInstalledListJson(packagesJson: string): string[] {
+    const packagesData = JSON.parse(packagesJson);
+    return packagesData.result.map((entry: any) => entry.SubscriberPackageName);
+  }
+
+  public async execute(
+    response: ContinueResponse<IsvDebugBootstrapConfig>
+  ): Promise<void> {
     const cancellationTokenSource = new vscode.CancellationTokenSource();
     const cancellationToken = cancellationTokenSource.token;
 
     const projectParentPath = response.data.projectUri;
     const projectPath = path.join(projectParentPath, response.data.projectName);
+    const projectMetadataTempPath = path.join(
+      projectPath,
+      this.relativeMetdataTempPath
+    );
+    const apexRetrievePackageXmlPath = path.join(
+      projectPath,
+      this.relativeApexPackageXmlPath
+    );
 
-    const createProjectExecution = new CliCommandExecutor(
+    // 1: create project
```

Comment:
If step 1 succeeds and a later step fails, should we delete the project?

Reply:
@vazexqi What are your thoughts? Should we clean up in case of errors or leave the project there for investigation purposes?

Reply:
Let's leave the project there for now.